virttest/qemu_devices/qdevices.py

```diff
@@ -1181,8 +1195,37 @@ def hotplug_hmp(self):
             out = "object_add %s" % _convert_args(self.params)
         return out
 
+    def _refresh_hotplug_props(self, params):
+        """
+        Refresh hotplug optional props as per params.
+
+        :return: A dict containing hotplug optional props.
+        """
+        return params
+
+    def _hotplug_qmp_mapping(self, qemu_version):
+        return self.hotplug_qmp if qemu_version in VersionInterval(
+            self.PROPS_DEPRECATED_VERSION_SCOPE) else self.hotplug_qmp_old
+
     def hotplug_qmp(self):
-        """ :return: the object-add command """
+        """ :return: the object-add command (since 5.0)"""
+        params = self.params.copy()
+
+        # qom-type and id are mandatory
+        kwargs = {
+            "qom-type": params.pop("backend"),
+            "id": params.pop("id")
+        }
+
+        # optional params
+        params = self._refresh_hotplug_props(params)
+        if len(params) > 0:
+            kwargs.update(params)
+
+        return "object-add", kwargs
+
+    def hotplug_qmp_old(self):
+        """ :return: the object-add command (before 5.0)"""
```

Comment:
Should we new a method for this? for example, if the behaviors of `object-add` are changed again since qemu-6.0, how could we name a method for it?
like following?
```
def hotplug_qmp(self): 
""" :return: the object-add command (since 6.0)"""

def hotplug_qmp_old(self):
""" :return: the object-add command (from 5.0 to 6.0)"""

def hotplug_qmp_older(self):
""" :return: the object-add command (before 5.0"""
```
I think we better not new a method for the new behaviors since we could not confirm these changes are last time, never changed forever.

Reply:
Use qemu version to differentiates the methods.